Remove implicit `object` from the base class

HarshCasper opened the pull request:
## Description 

This Pull Request removes implicit `object` from the Base Class. 

The Issue seems to be very minor but it improves the readability of the Code and since Python 3 inserts it for us behind the scenes, I suggested that its best remove them up. 

I'm a First-Time Contributor to Google Open-Source so it would be helpful for me understanding how PR workflow works and help me step up in future contributions as well.

afrozenator commented:
Hi @HarshCasper - Thanks for sending the PR, nice work! Can I request you to revert the files under trax/tf_numpy -- It's in the process of being moved to its own repo -- so we're not accepting PRs there at the moment.

HarshCasper commented:
Thanks for the kind words @afrozenator 

I have reverted all the files under `trax/tf_numpy` directory and squashed my Commits into one atomic commit. I guess it is ready to be merged now.

afrozenator commented:
Thanks @HarshCasper - working on getting this in now!
